code/backend/batch/utilities/helpers/DocumentProcessorHelper.py

```diff
@@ -55,12 +57,21 @@ def process(self, source_url: str, processors: List[Processor]):
                     documents: List[SourceDocument] = []
                     documents = document_loading.load(source_url, processor.loading)
                     documents = document_chunking.chunk(documents, processor.chunking)
-                    keys = list(map(lambda x: x.id, documents))
-                    documents = [
-                        document.convert_to_langchain_document()
-                        for document in documents
-                    ]
-                    return vector_store.add_documents(documents=documents, keys=keys)
+
+                    for document in documents:
+                        doc = {
+                            "id": document.id,
+                            "content": document.content,
+                            "content_vector": self.llm_helper.generate_embeddings(
+                                document.content
+                            ),
+                            # "metadata": document.metadata, this is just a full json escaped version of the entry
+                            "title": document.title,
+                            "source": document.source,
+                            "chunk": document.chunk,
+                            "offset": document.offset,
+                        }
+                        self.search_client.upload_documents([doc])
```

Comment:
Could you upload all of these at once?

Reply:
Great point, I'll do this and when/if we do this for real.